### PR TITLE
nixos/tika: init module

### PR DESCRIPTION
**Pull request by drupol**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/services/search/tika.nix [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether to open the firewall for Apache Tika.
+          This adds `services.tika.port` to `networking.firewall.allowedTCPPorts`.
+        '';
+      };
```
Comment (nim65s) [inline nixos/modules/services/search/tika.nix:51]: Is this implemented ? I would expect something like `networking.firewall.allowedTCPPorts = mkIf cfg.openFirewall [ cfg.settings.port ];` in `config = mkIf cfg.enable { … }`

Comment (drupol) [reply nixos/modules/services/search/tika.nix:51]: Yeah I forgot it, this is now fixed.

File: nixos/modules/services/search/tika.nix [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    systemd.services.tika = {
+      description = "Apache Tika Server";
+
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+
+      serviceConfig = {
+        Type = "simple";
+
+        ExecStart = "${getExe' cfg.package "tika-server"} --host ${cfg.listenAddress} --port ${toString cfg.port}";
```
Comment (nim65s) [inline nixos/modules/services/search/tika.nix:65]: `tika-server` is already in `meta.mainProgram`, so why not use `${getExe cfg.package}` instead ?

Comment (drupol) [reply nixos/modules/services/search/tika.nix]: Fixed. Note: Tika ships 2 binaries, `tika-server` is set as default. If by any chance, that is modified, the service will be broken now.

Comment (nim65s) [reply nixos/modules/services/search/tika.nix]: yes, but the test should catch this I think

Comment (drupol) [reply nixos/modules/services/search/tika.nix]: Oh yes definitely.

Comment (drupol) [reply nixos/modules/services/search/tika.nix]: anyway, I fixed it already.

**Review by GaetanLepage — APPROVED** [review]
LGTM !

**Pull request merged**